### PR TITLE
[refactor](nereids) Remove SlotBinder and FunctionBinder

### PR DESCRIPTION
**Pull request by 924060929**
The new ExpressionAnalyzer can do both bind slot and bind function, so I remove SlotBinder and FunctionBinder.

### CODE REVIEW

**Comment by 924060929** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java [hunk 1]
```diff
@@ -173,6 +173,9 @@ private void checkRootIsLimit(AnalyzedResult analyzedResult) {
     }
 
     private AnalyzedResult analyzeSubquery(SubqueryExpr expr) {
+        if (cascadesContext == null) {
+            throw new IllegalStateException("Missing CascadesContext");
+        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java:178]: check not null in ctor?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java [hunk 2]
```diff
@@ -124,14 +137,30 @@ public class ExpressionAnalyzer extends SubExprAnalyzer<ExpressionRewriteContext
     private boolean hasNondeterministic;
 
     /** ExpressionAnalyzer */
-    public ExpressionAnalyzer(Plan currentPlan, Scope scope, CascadesContext cascadesContext,
-            boolean enableExactMatch, boolean bindSlotInOuterScope) {
+    public ExpressionAnalyzer(Plan currentPlan, Scope scope,
+            @Nullable CascadesContext cascadesContext, boolean enableExactMatch, boolean bindSlotInOuterScope) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java:141]: why cascadesContext could be nullable?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java (hunk 1 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java:178]: In most of UTs, no ConnectContext the test thread. They only want to bind function, and not need to bind Subquery/Slot, and the CascadesContext is not needed.

So we can not check In the constructor.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java (hunk 2 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java:141]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java (hunk 1 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java:178]: Bind slot/subquery: need context
Bind function: not need context

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**